src/OpenTelemetry/Internal/CircularBuffer.cs

```diff
@@ -186,5 +205,11 @@ public T Read()
                 return value;
             }
         }
+
+        private static bool DetectShouldYieldWhileSpinning()
+        {
+            var isRunningOnSingleCore = Environment.ProcessorCount == 1;
+            return isRunningOnSingleCore;
```

Comment:
this may not me the actual number of cores available to *this* process.

Reply:
https://docs.microsoft.com/en-us/dotnet/api/system.threading.spinwait?view=net-6.0#remarks It might be best to switch to the framework provided mechanisms.